src/commands/general.rs

```diff
@@ -635,3 +637,203 @@ pub async fn help(
     }
     Ok(())
 }
+
+/// Votemute an User
+///
+/// When enough regulars vote to mute a user the user gets muted
+/// ``votemute [user]``
+#[poise::command(prefix_command, slash_command, guild_only, category = "General")]
+pub async fn votemute(
+    ctx: Context<'_>,
+    #[rename = "user"]
+    #[description = "User to votemute"]
+    mut target_user_m: Member,
+) -> Result<(), Error> {
+    const TIMEOUT_DURATION: usize = 30;
+    // pub votemute_users: Mutex<HashMap<UserId, (i64, Vec<UserId>)>>,
+    // HashMap<Target, (Timestamp; List of Users that voted)>
+
+    ctx.defer().await?;
```

Comment:
Calling this before doing all the checks that may result in an ephemeral error will cause them to not be ephemeral, should be located after the checks before the actual logic.